sync.go

```diff
@@ -42,6 +42,10 @@ var (
 	disabled = func(args ...interface{}) string { return fmt.Sprint(args...) }
 )
 
+// Running too many syncs at once can exhaust file descriptor limits.
+// Empirically, ~90 is enough to hit the macOS default limit of 256.
+const maxConcurrentSyncs = 10
```

Comment:
Ah, yeah, this covers the network downloading too so it probably does need to be higher (although since the lion's share of this traffic goes to github it's kind of antisocial to run so many downloads in parallel to try to speed things up). The problematic case for us is that when almost everything is up to date and one thing changed, the concurrent `git rev-parse` calls cause it to fail, but for when we are downloading things we need concurrency independent of the number of CPUs (and in that case the limit may be somewhat lower than 90 since the rev-parse calls finish quickly). I'll change it to 25.